torch/_dynamo/variables/builder.py

```diff
@@ -419,9 +437,16 @@ def index_source(key):
                 else:
                     return key
 
+            # If keys_are_types_or_namedtuple is True, don't guard on dictionary values.
+            #
+            # In this case, we are guarding on the dictionary version (i.e. it wasn't
```

Comment:
While you can ignore guards here because you are guarding elsewhere - I strongly dislike the pattern that `ignore_guards` creates. It's a huge hammer to just be able to pass like this, imo. I would rather we build `result` differently, or build it and then strip the guards off.

Reply:
I agree. So, in order not to leave `ignore_guards`, I will specialize a case for `SUPPORTED_NODES`.